src/main/java/com/denizenscript/clientizen/scripts/containers/gui/elements/DynamicLabelElement.java

```diff
@@ -0,0 +1,68 @@
+package com.denizenscript.clientizen.scripts.containers.gui.elements;
+
+import com.denizenscript.clientizen.scripts.containers.gui.GuiScriptContainer;
+import com.denizenscript.denizencore.objects.core.ColorTag;
+import com.denizenscript.denizencore.tags.ParseableTag;
+import com.denizenscript.denizencore.tags.TagContext;
+import com.denizenscript.denizencore.tags.TagManager;
+import com.denizenscript.denizencore.utilities.YamlConfiguration;
+import com.denizenscript.denizencore.utilities.debugging.Debug;
+import io.github.cottonmc.cotton.gui.widget.WDynamicLabel;
+import io.github.cottonmc.cotton.gui.widget.WWidget;
+import io.github.cottonmc.cotton.gui.widget.data.HorizontalAlignment;
+
+import static com.denizenscript.clientizen.scripts.containers.gui.GuiScriptContainer.getDebugPath;
+import static com.denizenscript.clientizen.scripts.containers.gui.GuiScriptContainer.getTaggedEnum;
+import static com.denizenscript.clientizen.scripts.containers.gui.GuiScriptContainer.getTaggedObject;
+
+public class DynamicLabelElement implements GuiScriptContainer.GuiElementParser {
+
+    // <--[language]
+    // @name Dynamic Label GUI Element
+    // @group GUI System
+    // @description
+    // Dynamic labels work similarly to <@link language Label GUI Element>s, but with dynamically updating text; they have a UI type of "dynamic_label".
+    //
+    // <code>
+    // ui_type: dynamic_label
+    // # The dynamically updating text for the label, required.
+    // text: <text>
+    // # The horizontal alignment for the label's text, optional.
+    // horizontal_alignment: LEFT/CENTER/RIGHT
+    // # The color for the label's text, optional.
+    // color: <ColorTag>
+    // </code>
+    // The text is parsed every time the label is rendered, which means tags in it are constantly read.
+    // This is fine for simple tags, but in the case of highly complex ones it may be better to implement
+    // your own system for updating the text using a normal label, to only do it when relevant.
```

Comment:
The text, including tags, is parsed every frame. With computationally expensive tags, it might prove better to use your own system for updating the text.

Reply:
Done, but Ayaed it a bit to fit other meta around it & added a line on why just in case.